src/server/req_rerun.c

```diff
@@ -285,16 +271,21 @@ req_rerunjob(struct batch_request *preq)
 		/* Setting deleted subjobs count to 0,
 		 * since all the deleted subjobs will be moved to Q state
 		 */
-		parent->ji_ajtrk->tkm_dsubjsct = 0;
-
-		for (i=0; i<parent->ji_ajtrk->tkm_ct; i++) {
-			if ((pjob = parent->ji_ajtrk->tkm_tbl[i].trk_psubjob)) {
-				if (check_job_state(pjob, JOB_STATE_LTR_RUNNING))
+		parent->ji_ajinfo->tkm_dsubjsct = 0;
+
+		for (i = parent->ji_ajinfo->tkm_start; i <= parent->ji_ajinfo->tkm_end; i += parent->ji_ajinfo->tkm_step) {
+			pjob = get_subjob_state(parent, i, &sjst, NULL);
+			if (sjst == JOB_STATE_LTR_UNKNOWN)
+				continue;
+			if (pjob) {
+				if (sjst == JOB_STATE_LTR_RUNNING)
 					dup_br_for_subjob(preq, pjob, req_rerunjob2);
-				else
+				else {
 					force_reque(pjob);
+					job_purge(pjob);
+				}
```

Comment:
Looks like missed to remove this line, I will remove it, we should purge job here on re-queue (it will be in create_subjob when next time subjob runs)